src/nested/Car.java

```diff
@@ -0,0 +1,66 @@
+package nested;
+
+public class Car {
+    String colour;
+    int doorCount;
+    Engine engine;
+    GasTank gasTank;
+
+    public Car(String colour, int doorCount, Engine engine, int value) {
+        this.colour = colour;
+        this.doorCount = doorCount;
+        this.engine = engine;
+        this.gasTank = this.new GasTank(value);
+    }
+
+    @Override
+    public String toString() {
+        return "Car{" +
+                "colour='" + colour + '\'' +
+                ", doorCount=" + doorCount +
+                ", engine=" + engine +
+                ", gasTank=" + gasTank +
+                '}';
+    }
+
+    public static class Engine { // static Nested class
```

Comment:
static? there is no reason to make it static))) prove the usability of nested static class in code